ocaml/xapi/repository.ml

```diff
@@ -737,40 +737,29 @@ let apply_updates' ~__context ~host ~updates_info ~livepatches ~acc_rpm_updates
       ) ;
   (* Evaluate recommended/pending guidances *)
   let recommended_guidances, pending_guidances =
-    let new_recommended_gs =
+    let guidances =
       (* EvacuateHost will be applied before applying updates *)
       eval_guidances ~updates_info ~updates:acc_rpm_updates ~kind:Recommended
         ~livepatches:successful_livepatches ~failed_livepatches
       |> List.filter (fun g -> g <> Guidance.EvacuateHost)
     in
-    let recommended_guidances' =
+    let guidances' =
       merge_with_unapplied_guidances ~__context ~host ~kind:Recommended
-        ~guidances:new_recommended_gs
+        ~guidances
```

Comment:
Sorry, I think there is one debt from my previous code change:
`merge_with_unapplied_guidances` should be done after the `guidances` is updated by failed livepatch. Otherwise there might be duplicated `Guidance.RebootHostOnLivePatchFailure` guidance.